src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8359,21 +8253,14 @@ private bool IsPossibleNewExpression()
                 return !typedIdentifier.Value;
             }
 
-            var resetPoint = this.GetResetPoint();
-            try
-            {
-                // skips new keyword
-                EatToken();
+            using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
 
-                ScanTypeFlags st = this.ScanType();
+            // skips new keyword
+            EatToken();
+            ScanTypeFlags st = this.ScanType();
 
-                return !IsPossibleMemberName() || st == ScanTypeFlags.NotType;
-            }
-            finally
-            {
-                this.Reset(ref resetPoint);
-                this.Release(ref resetPoint);
-            }
+            // TODO(cyrusn): Why not do the `st == ScanTypeFlags.NotType` check first?
```

Comment:
It's disallowed to add TODO comments. Please file a tracking issue instead

Reply:
removed.  adding to the backlog of issues doesn't seem helpful here as i can't imagine it ever being looked at.